web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -0,0 +1,357 @@
+      }
+    })
+  }
+
+  /**
+   * This function scales the x-axis and y-axis and initializes empty focus 
+   * and brush elements.
+   * 
+   * Structure of the SVG:
+   * <svg>
+   *  <g> 
+   *    // Top bar chart (focus).
+   *    <g>
+   *      // Top x-axis (xAxisFocus).
+   *    </g>
+   *    <g>
+   *      // Top y-axis (yAxis).
+   *    </g>
+   *    <text>
+   *      // Top y-axis title.
+   *    </text>
+   *  </g>
+   *  <g>
+   *    // Bottom bar chart (brush).
+   *    <g>
+   *      // Bottom x-axis (xAxisBrush).
+   *    </g>
+   *  </g>
+   */
+  private initialChart(): void {
+    const element = this.barChartContainer.nativeElement;
+    const elementWidth = element.clientWidth;
+    const elementHeight = element.clientHeight;
+    const marginFocus = { top: 60, right: 90, bottom: 150, left: 70 };
+    const marginBrush = { top: 370, right: 60, bottom: 20, left: 60};
+
+    this.heightFocus = elementHeight;
+    this.heightBrush = 30;
+    this.width = elementWidth;
+
+    this.svg = (d3
+      .select(element)
+      .append('svg') as d3SVG)
+      .attr('width', elementWidth)
+      .attr('height', elementHeight);
+
+    this.focus = this.svg
+      .append('g')
+      .attr('transform','translate(0, 0)');
+    this.brush = this.svg
+      .append('g')
+      .attr('transform', `translate(0, ${marginBrush.top})`);
+
+    this.xScaleFocus = d3
+      .scaleBand()
+      .rangeRound([marginFocus.left, elementWidth - marginFocus.right])
+      .padding(0.1);
+    this.yScaleFocus = d3
+      .scaleLinear().range([elementHeight - marginFocus.bottom, marginFocus.top]);
+    this.xAxisFocus = this.focus
+      .append('g')
+      .attr('class', 'axis axis--x')
+      .attr('transform', `translate(0, ${elementHeight - marginFocus.bottom})`);
+    this.yAxis = this.focus
+      .append('g')
+      .attr('transform', `translate(${marginFocus.left}, 0)`)
+      .style('color', BarChartComponent.COLOR_LIGHT_GRAY);
+
+    this.focus.append("text")
+      .attr("text-anchor", "middle") 
+      .attr("transform", 
+        "translate("+(marginFocus.left/2)+","+
+        ((elementHeight-marginFocus.bottom+marginFocus.top)/2)+")rotate(-90)")
+      .attr("fill", BarChartComponent.COLOR_LIGHT_GRAY)
+      .text("Duration of the push (hours)")
+      .style("font", "12px sans-serif");
+
+    this.xScaleBrush = d3
+      .scaleBand()
+      .rangeRound([marginBrush.left, elementWidth - marginBrush.right])
+      .padding(0.1)
+    this.yScaleBrush = d3
+      .scaleLinear().range([this.heightBrush, 0]);
+    this.xAxisBrush = this.brush
+      .append('g')
+      .attr('transform', `translate(0, ${this.heightBrush})`);
+  }
+
+  /**
+   * This function updates the focus barchart based on the data of the 
+   * dropdown selection.
+   * 
+   * @param dataSelection: Value selected for the bar chart
+   */
+  public updateChart(dataSelection: number): void {
+    const dataSelected = (dataSelection === 0) ? this.dataAll : this.dataComplete;
```

Comment:
How about making constants for selecting data all and complete instead of just comparing with `0`?

Reply:
This depends on the value that the drop down menu will output. I prefer to leave this to the next PR to determine whether it is better for number or data values :)

Reply:
Yes that's true. But I meant instead of comparing with `0`, we should compare with a constant.

Reply:
I see. It is fixed!